src/plugins/topic_viewer/TopicViewer.cc

```diff
@@ -0,0 +1,282 @@
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_FLOAT);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_INT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT32);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_UINT64);
+  this->plotableTypes.push_back(FieldDescriptor::Type::TYPE_BOOL);
+
+  this->CreateModel();
+
+  ignition::gui::App()->Engine()->rootContext()->setContextProperty(
+                "TopicsModel", this->dataPtr->model);
+}
+
+//////////////////////////////////////////////////
+TopicViewer::~TopicViewer()
+{
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::LoadConfig(const tinyxml2::XMLElement *)
+{
+  if (this->title.empty())
+    this->title = "Topic Viewer";
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::CreateModel()
+{
+  this->dataPtr->model = new TopicsModel();
+
+  std::vector<std::string> topics;
+  this->dataPtr->node.TopicList(topics);
+
+  for (unsigned int i = 0; i < topics.size(); ++i)
+  {
+    std::vector<ignition::transport::MessagePublisher> infoMsgs;
+    this->dataPtr->node.TopicInfo(topics[i], infoMsgs);
+    std::string msgType = infoMsgs[0].MsgTypeName();
+    this->AddTopic(topics[i], msgType);
+  }
+}
+
+//////////////////////////////////////////////////
+QStandardItemModel *TopicViewer::Model()
+{
+    return reinterpret_cast<QStandardItemModel *>(this->dataPtr->model);
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::AddTopic(const std::string &_topic,
+                           const std::string &_msg)
+{
+  QStandardItem *topicItem = this->FactoryItem(_topic, _msg);
+  topicItem->setWhatsThis("Topic");
+  QStandardItem *parent = this->dataPtr->model->invisibleRootItem();
+  parent->appendRow(topicItem);
+
+  this->AddField(topicItem , _msg, _msg);
+}
+
+//////////////////////////////////////////////////
+void TopicViewer::AddField(QStandardItem *_parentItem,
+                           const std::string &_msgName,
+                           const std::string &_msgType)
+{
+  QStandardItem *msgItem;
+
+  // check if it is a topic, to skip the extra level of the topic Msg
+  if (_parentItem->whatsThis() == "Topic")
+  {
+    msgItem = _parentItem;
+    // make it different, so next iteration will make a new msg item
+    msgItem->setWhatsThis("Msg");
+  }
+  else
+  {
+    msgItem = this->FactoryItem(_msgName, _msgType);
+    _parentItem->appendRow(msgItem);
+  }
+
+  auto msg = ignition::msgs::Factory::New(_msgType);
+  if (!msg)
+      return;
+
+  auto msgDescriptor = msg->GetDescriptor();
+  if (!msgDescriptor)
+  {
+    ignwarn << "Null Descriptor of Msg: " << _msgType << std::endl;
+    return;
+  }
+
+  for (int i = 0 ; i < msgDescriptor->field_count(); ++i)
+  {
+    auto msgField = msgDescriptor->field(i);
+
+    if (msgField->is_repeated())
+        continue;
```

Comment:
```suggestion
      continue;
```
use two space indentation